frontend/src/metabase/search/components/SearchResult.tsx

```diff
@@ -150,7 +150,13 @@ export function SearchResult({
           )}
           <Score scores={result.scores} />
         </div>
-        {loading && <ResultSpinner size={24} borderWidth={3} />}
+        {loading && (
+          <ResultSpinner
+            data-testid="search-result-loading-spinner"
```

Comment:
Ah yeah so this is a weird one. Since these `SearchResult` elements have a `data-testid="loading-spinner"`, they break every test when we're using `waitFor(<data-testid="loading-spinner" to be removed>)` when we're trying to check when the `Search.ListLoader` is finished loading. So, we're basically helping the test differentiate between "SearchApp is finished loading" and "SearchResult is finished loading".

Reply:
I got your point. I think my point is that `search-result-loading-spinner` would not be obvious to others why this is needed because it's not asserted anywhere. I think this is worth having a comment, otherwise, it would be easily mistaken as not being used and removed without knowing it's gonna break tests.

Reply:
Ah ok sounds good, I'll add a small comment there